src/crypto/secretstream/secretstream_macros.rs

```diff
@@ -297,7 +302,12 @@ impl Stream<Pull> {
         let (ad_p, ad_len) = ad
             .map(|ad| (ad.as_ptr(), ad.len()))
             .unwrap_or((ptr::null(), 0));
-        let mut tag: u8 = unsafe { mem::uninitialized() };
+        let mut tag = {
+            // Tag can be one of Message=0, Push=1, Rekey=2, Final=3 states
+            // So even if tag is not initialized in $pull_name call
+            //  we will get an error in Tag::from_u8 which is better than UB
+            u8::max_value()
+        };
```

Comment:
This is less unsafe but you just remove unsafe and make the code more slow. It's a choice but I think you mix the clear goal to replace `uninitialized` by `MaybeUninit` and the goal to remove unsafe at all cost. This should be in a separate PR I think.

Reply:
Agreed, just use `MaybeUninit` here.

Reply:
Fixed